### PR TITLE
Move expiration checks for Root to the right place

### PR DESCRIPTION
**Pull request by awwad**
Root metadata must be verified even if it is expired.  This is
important for root chaining.  It is only necessary to check
whether or not Root is expired when we are done updating Root,
and we want to update the other top-level roles.  At that point,
if our currently-trusted Root metadata is expired, we MUST stop
and only continue once we have unexpired Root metadata.

Consequently, this PR:
- kills Root expiration checks in `_verify_uncompressed_metadata_file`
- adds a Root expiration check in `refresh()`, before we try to
  update the other top-level roles
- adds some TODOs suggesting other expiration checks to exclude
- adds a TODO suggesting the migration to the new specification's
  update-root-first-and-don't-check-root-against-snapshot-hashes
  model.

### CODE REVIEW

**Comment by awwad** [comment]
- [x] local upTUF tests proceed as expected
- [x] local Uptane tests succeed when using this upTUF branch
- [x] [Uptane CI tests](https://github.com/uptane/uptane/pull/182) succeed when using this upTUF branch
- [x] Uptane demo (demo/demo_single_command_run.py) runs as expected when using this upTUF branch

File: tuf/client/updater.py [hunk 1]
```diff
@@ -1400,6 +1402,14 @@ def refresh(self, unsafely_update_root_if_necessary=True):
         logger.info('An expired Root metadata was loaded and must be updated.')
         raise
 
+
+    # If after trying to update Root, our currently-trusted Root metadata is
+    # *still* expired, then we must abort.
+    # We must not use expired Root metadata to verify other top-level roles.
+    # Hold out for a non-expired Root.
+    self._ensure_not_expired(self.metadata['current']['root'], 'root')
```
Comment (lukpueh) [inline tuf/client/updater.py:1410]: I suppose you only have to check again if you did update, so you could move this after (and on the same indentation level as) L1398.

**Review by lukpueh — COMMENTED** [review]
I don't fully understand the consequences from excluding expiration check in `_verify_uncompressed_metadata_file`, but the other changes look sensible. Also, I ran the tests locally  and they pass.

File: tuf/client/updater.py (hunk 1 above)
Comment (awwad) [reply tuf/client/updater.py:1410]: Oh, quite true.  Thanks!

**Pull request merged**